src/textinput/TextInput.ts

```diff
@@ -0,0 +1,64 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { FormLabelMixin, FormLabelMixinProperties } from '@dojo/widget-core/mixins/FormLabel';
+import { v } from '@dojo/widget-core/d';
+import * as css from './styles/textinput.css';
+
+/**
+ * @type TextInputProperties
+ *
+ * Properties that can be set on a TextInput component
+ *
+ * @property type			Input type, e.g. text, email, tel, etc.
+ * @property onInput	Called when the node's 'input' event is fired
+ * @property onChange	Called when the node's 'change' event is fired
+ * @property onFocus	Called when the input is focused
+ * @property onBlur		Called when the input loses focus
+ */
+export interface TextInputProperties extends ThemeableProperties, FormLabelMixinProperties {
+	type?: string;
+	onInput?(event: Event): void;
```

Comment:
Is this an extensive list of all of the events that can be attached? dojo1 textbox dijit has `keyPress`, `keyDown` etc..